bluemira/equilibria/coils/_coil.py

```diff
@@ -0,0 +1,521 @@
+            idx = CoilNumber.__PF_counter
+            CoilNumber.__PF_counter += 1
+
+        return idx
+
+
+class Coil(CoilFieldsMixin):
+    """
+    Coil Object
+
+    For used with PF/CS/passive coils
+
+    Parameters
+    ----------
+    x: float
+        Coil geometric centre x coordinate [m]
+    z: float
+        Coil geometric centre z coordinate [m]
+    dx: float
+        Coil radial half-width [m] from coil centre to edge (either side)
+    dz: float
+        Coil vertical half-width [m] from coil centre to edge (either side)
+    name: str
+        The name of the coil
+    ctype: Union[str, CoilType]
+        Type of coil as defined in CoilType
+    current: float (default = 0)
+        Coil current [A]
+    j_max: float
+        Maximum current density in the coil [A/m^2]
+    b_max: float
+        Maximum magnetic field at the coil [T]
+    discretisation: float
+        discretise the coil (between 0 and 1)
+    n_turns: int
+        Number of turns
+
+    """
+
+    def __init__(
+        self,
+        x: float,
+        z: float,
+        dx: float = None,
+        dz: float = None,
+        name: Optional[str] = None,
+        ctype: Union[str, CoilType] = CoilType.NONE,
+        current: float = 0,
+        j_max: float = np.nan,
+        b_max: float = np.nan,
+        discretisation: float = 1,
+        n_turns: int = 1,
+    ):
+        self._dx = None
+        self._dz = None
+        self._discretisation = 1
+        self._flag_sizefix = None not in (dx, dz)
+
+        self.x = x
+        self.z = z
+        self.dx = dx
+        self.dz = dz
+        self.discretisation = discretisation
+        self.current = current
+        self.j_max = j_max
+        self.b_max = b_max
+        self.ctype = ctype
+        self.name = name
+        self.n_turns = n_turns
+
+        self._number = CoilNumber.generate(self.ctype)
+        if self.name is None:
+            self.name = f"{self._ctype.name}_{self._number}"
+
+        # check if dx and not dz set
+        # check of j max set
+        self._validate_size()
+        if not self._flag_sizefix and None in (self.dx, self.dz):
+            self._dx, self._dz = 0, 0
+            self._discretise()
+            self._set_coil_attributes()
+
+    def __repr__(self):
+        """
+        Pretty printing
+        """
+        return (
+            f"{type(self).__name__}({self.name} ctype={self.ctype.name} x={self.x:.2g}"
+            f" z={self.z:.2g} dx={self.dx:.2g} dz={self.dz:.2g} current={self.current:.2g}"
+            f" j_max={self.j_max:.2g} b_max={self.b_max:.2g}"
+            f" discretisation={self.discretisation:.2g})"
+        )
+
+    def plot(self, ax=None, subcoil=True, label=False, force=None, **kwarg):
+        """
+        Plot a Coil
+        """
+        return CoilGroupPlotter(
+            self, ax=ax, subcoil=subcoil, label=label, force=force, **kwarg
+        )
+
+    def n_coils(self):
+        """Number of coils in coil"""
+        return 1
```

Comment:
Because coils and coilgroups can be nested next to each other I thought it was important to be able to access the number of coils in both instances. Will add a note